### PR TITLE
Deprecate scheduler traits v1 and v2

### PR DESCRIPTION
**Pull request by chungquantin**
This PR add `#[deprecated]` attribute to v1 and v2 of the schedule trait. Proposed in this issue: https://github.com/paritytech/polkadot-sdk/issues/3676 

```rust
#[allow(deprecated)]
#[deprecated = "traits::schedule::v1 is deprecated. Please use v3 instead."]
pub mod v1 {
...
}

#[allow(deprecated)]
#[deprecated = "traits::schedule::v2 is deprecated. Please use v3 instead."]
pub mod v2 {
...
}
```

polkadot address: 19nSqFQorfF2HxD3oBzWM3oCh4SaCRKWt1yvmgaPYGCo71J

### CODE REVIEW

File: substrate/frame/support/src/traits/schedule.rs [hunk 1]
```diff
@@ -130,7 +130,8 @@ impl<T: Decode, H> MaybeHashed<T, H> {
 	}
 }
 
-// TODO: deprecate
+#[allow(deprecated)]
+#[deprecated = "traits::schedule::v1 is deprecated. Please use v3 instead."]
```
Comment (ggwpez) [inline substrate/frame/support/src/traits/schedule.rs:134]: There is some uniform grammar for this:  
https://github.com/paritytech/polkadot-sdk/blob/6b5995ffd162d88df1e3baeaafe05869d1a7d966/docs/contributor/DEPRECATION_CHECKLIST.md#L25

File: substrate/frame/support/src/traits/schedule.rs [hunk 2]
```diff
@@ -130,7 +130,8 @@ impl<T: Decode, H> MaybeHashed<T, H> {
 	}
 }
 
-// TODO: deprecate
+#[allow(deprecated)]
```
Comment (ggwpez) [inline substrate/frame/support/src/traits/schedule.rs:133]: Do you really need those?  

I think it should rather go on the implementation of any of the traits.

Comment (chungquantin) [reply substrate/frame/support/src/traits/schedule.rs]: I have added a commit to resolve your feedback

**Comment by chungquantin** [comment]
@ggwpez resolved to follow the convention.

**Comment by ggwpez** [comment]
Yes no please put the `allow(deprecated)` onto the parts that actually need it, like `pallet-scheduler`.  
CI should show it.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5553090

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5553109

File: substrate/frame/support/src/traits/schedule.rs [hunk 3]
```diff
@@ -130,7 +130,7 @@ impl<T: Decode, H> MaybeHashed<T, H> {
 	}
 }
 
-// TODO: deprecate
+#[deprecated(note = "Use `v3` instead")]
```
Comment (ggwpez) [inline substrate/frame/support/src/traits/schedule.rs:133]:
```suggestion
#[deprecated(note = "Use `v3` instead. Will be removed after September 2024.")]
```

File: substrate/frame/support/src/traits/schedule.rs [hunk 4]
```diff
@@ -266,27 +275,31 @@ pub mod v1 {
...
 
 		fn next_dispatch_time(id: Vec<u8>) -> Result<BlockNumber, ()> {
+			#[allow(deprecated)]
 			T::next_dispatch_time(id)
 		}
 	}
 }
 
-// TODO: deprecate
+#[deprecated(note = "Use `v3` instead")]
```
Comment (ggwpez) [inline substrate/frame/support/src/traits/schedule.rs:302]:
```suggestion
#[deprecated(note = "Use `v3` instead. Will be removed after September 2024.")]
```

File: substrate/frame/scheduler/src/lib.rs [hunk 5]
```diff
@@ -1435,6 +1435,8 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
+// TODO: migrate `schedule::v2::Anon` to `v3`
```
Comment (ggwpez) [inline substrate/frame/scheduler/src/lib.rs:1438]:
```suggestion
```
I know that there were some TODOs in the code before, but its not really good.  
Will create another issue to remove these traits in the future.

Comment (chungquantin) [reply substrate/frame/scheduler/src/lib.rs:1438]: Gotcha!

**Review by ggwpez — APPROVED** [review]

**Comment by chungquantin** [comment]
@ggwpez I have updated the prdoc, could you help me to review again?

**Comment by ggwpez** [comment]
Yea LGTM, but it needs reviews from other ppl as well. Dont know where you could best ask, maybe in the fellowship public channel: https://matrix.to/#/#fellowship-open-channel:parity.io

Normally i would ping it internally to colleagues, but that does not scale well.

**Review by liamaharon — APPROVED** [review]

**Pull request merged**